api/actions/apply_manifest.go

```diff
@@ -64,7 +64,7 @@ func (a *ApplyManifest) checkAndUpdateDefaultRoute(ctx context.Context, authInfo
 		return nil
 	}
 
-	existingRoutes, err := a.routeRepo.ListRoutesForApp(ctx, authInfo, appRecord.SpaceGUID, appRecord.GUID)
+	existingRoutes, err := a.routeRepo.ListRoutesForApp(ctx, authInfo, appRecord.GUID, appRecord.SpaceGUID)
```

Comment:
Is there any test that could have caught this?

Reply:
We do not, both `e2e` and `integration` tests have cases that always satisfies [this condition](https://github.com/cloudfoundry/cf-k8s-controllers/blob/d789b4b222940e961f9864b05b93317dab3d0db1/api/actions/apply_manifest.go#L63) and returning back. 
I'll add a new test cases for this scenario

Reply:
okay, I have added an e2e test that beats the condition above and tests this code path.

Reply:
The e2e probably doesn't hurt but this is the typical mistake that should have been caught by unit tests. The unit test for this action never checks arguments for mocked methods though 😞  which is something we should definitely fix.